### PR TITLE
Implement Dashboard Feature Enhancements

### PR DESCRIPTION
**Pull request by Fayroth**
This pull request introduces significant enhancements to the dashboard feature, focusing on widget management and user interface improvements. The following changes have been made:

- **Dashboard Screen:** Implemented the ability to reorder widgets on the dashboard screen, allowing users to arrange their widgets as desired.
- **UI Enhancements:** Upgraded the UI colors to align with the Figma design. Please note that these changes are temporary while we await the update of Theme.kt.
- **ViewModel:** Updated the DashboardViewModel to support the new widget reordering functionality, ensuring seamless state management.
- **Repository:** Enhanced repository functions to manage the addition, removal, and reordering of widgets, ensuring data persistence.
- **Widget Data Class:** Modified the Widget data class to include a new field, order, which retains widget positions saved in the database.
- **Testing:** Achieved over 80% code coverage through comprehensive unit and UI tests, ensuring robust functionality.

**Testing:**

- All new features have been tested, and UI interactions have been validated using automated tests.
- Notably, I encountered challenges when testing the drag gesture for reordering widgets. The drag-and-drop experience is not as smooth as expected, and I plan to conduct further research to improve this functionality.

**Related Issues:**

- N/A

**Notes:**

- Please review the changes and provide any feedback.
- The new functionality is ready for review and can be merged upon approval.

### CODE REVIEW

**Comment by Fayroth** [comment]
I just saw that it was not covered at 80% on new code, im working on it

**Comment by abj10** [comment]
The code is well-written and aligns with the functionality required for reordering widgets as described in the user stories. Everything looks good to me (LGTM).

File: app/src/androidTest/java/com/github/se/eduverse/ui/DashboardScreenTest.kt
Comment (mehdixboutaleb) [inline app/src/androidTest/java/com/github/se/eduverse/ui/DashboardScreenTest.kt:1]: The test function cover great features, but I think it should also test a successful drag and drop operation, the empty state and the adding/removal of a widget even though some of these functions are tested in the repository, I think it should be also tested here as these are UI behaviors

File: app/src/main/java/com/github/se/eduverse/model/Widget.kt
Comment (mehdixboutaleb) [inline app/src/main/java/com/github/se/eduverse/model/Widget.kt:1]: Looks good to me

File: app/src/main/java/com/github/se/eduverse/repository/DashboardRepository.kt [hunk 1]
```diff
@@ -34,15 +37,25 @@ class DashboardRepositoryImpl(private val firestore: FirebaseFirestore) : Dashbo
                 }
             awaitClose { listener.remove() }
           }
-          .catch { emit(emptyList()) } // Catch any exceptions and emit an empty list
+          .catch { emit(emptyList()) }
```
Comment (mehdixboutaleb) [inline app/src/main/java/com/github/se/eduverse/repository/DashboardRepository.kt:40]: You might want to log errors in your catch block to ensure you can debug issues effectively.

File: app/src/main/java/com/github/se/eduverse/repository/DashboardRepository.kt
Comment (mehdixboutaleb) [inline app/src/main/java/com/github/se/eduverse/repository/DashboardRepository.kt:1]: Apart from the empty catch block, nothing to say

File: app/src/main/java/com/github/se/eduverse/ui/dashboard/DashboardScreen.kt
Comment (mehdixboutaleb) [inline app/src/main/java/com/github/se/eduverse/ui/dashboard/DashboardScreen.kt:1]: The use of a function here for the reordable dashboard is clever as it makes the code cleaner and easier to read

File: app/src/main/java/com/github/se/eduverse/viewmodel/DashboardViewModel.kt
Comment (mehdixboutaleb) [inline app/src/main/java/com/github/se/eduverse/viewmodel/DashboardViewModel.kt:1]: Looks good for me

File: app/src/test/java/com/github/se/eduverse/repository/DashboardRepositoryTest.kt
Comment (mehdixboutaleb) [inline app/src/test/java/com/github/se/eduverse/repository/DashboardRepositoryTest.kt:1]: Clear and concise nothing to say

**Review by mehdixboutaleb — COMMENTED** [review]
Overall the code is clear and nice to read, some missing test in the DashboardScreenTest, but a nice implementation focusing on new features

**Review by abj10 — APPROVED** [review]
The code is well-written and aligns with the functionality required for reordering widgets as described in the user stories. Everything looks good to me (LGTM).
